bson/decoder.go

```diff
@@ -145,3 +145,40 @@ func (d *Decoder) DefaultDocumentM() {
 func (d *Decoder) DefaultDocumentD() {
 	d.defaultDocumentD = true
 }
+
```

Comment:
Should we add a Decode setter for "defaultDocumentType" and deprecate 

- `DecodeContext.DefaultDocumentM`
- `DecodeContext.DefaultDocumentD`

Reply:
Yes, we should deprecate `DecodeContext.DefaultDocumentM` and `DecodeContext.DefaultDocumentD` because setting configurations on a `DecodeContext` is unnecessary in Go Driver 2.0. However, we should stick with the existing API for setting `DefaultDocumentD` or `DefaultDocumentM` on the `Decoder`.

Also, keep in mind we're planning to remove `DefaultDocumentD` from `Decoder` in Go Driver 2.0 because that will be the default behavior, so there's no need for a configuration method that sets the default behavior. We can't deprecate it because we don't expect users to stop using it in Go Driver 1.x (i.e. while the "ancestor" behavior is the default), but we are planning to remove it.